gtest_parallel.py

```diff
@@ -398,9 +436,20 @@ def record_test_time(self, binary, testname, runtime_ms):
   def write_to_file(self, save_file):
     "Write all the times to file."
     try:
-      with open(save_file, "wb") as f:
-        with gzip.GzipFile("", "wb", 9, f) as gzf:
-          cPickle.dump(self.__times, gzf, cPickle.HIGHEST_PROTOCOL)
+      with TestTimes.LockedFile(save_file, 'a+b') as fd:
+        fd.seek(0)
+        try:
+          with gzip.GzipFile(fileobj=fd, mode='rb') as gzf:
+            times = cPickle.load(gzf)
+
+          times.update(self.__times)
+        except (EOFError, IOError, cPickle.UnpicklingError, zlib.error):
```

Comment:
Looking at what can be raised during unzipping/unpickling why not just catch either everything (with bare `except`) or catch generic `Exception`? Documentation is rather vague for both modules. In particular, gzip doesn't specify what can be raised and cPickle.load can raise a lot more than currently we have in `except`.